modules/vaos/app/controllers/vaos/v2/appointments_controller.rb

```diff
@@ -62,6 +62,25 @@ def create
         render json: { data: serialized }, status: :created
       end
 
+      def create_draft
+        # TODO: validate referral_id from the cache from prior referrals response,
+        # TODO: validate that the referral doesn't already have a confirmed appointment #
+        # TODO: cache provider_id, appointment_type_id, end_date from prior referrals response and use here
+        draft_appointment = eps_appointment_service.create_draft_appointment(referral_id: draft_params[:referral_id])
+
+        provider = eps_provider_service.get_provider_service(provider_id: draft_params[:provider_id])
+
+        response_data = OpenStruct.new(
+          id: draft_appointment.id,
+          provider: provider,
+          slots: fetch_provider_slots,
+          drive_time: fetch_drive_times(provider)
+        )
+
+        serialized = Eps::DraftAppointmentSerializer.new(response_data)
+        render json: serialized, status: :created
```

Comment:
What if one of the API call fails? Can we return `:multistatus` as the current `AppointmentsController` endpoint?  Please ignore, if this is planned in future PR.

Reply:
:multistatus doesn't apply here. This is 1 resource that is returned, not an array of resources.